internal/internal_update.go

```diff
@@ -253,6 +253,8 @@ func defaultUpdateHandler(
 		return
 	}
 	scheduler.Spawn(ctx, name, func(ctx Context) {
+		defer getState(ctx).dispatcher.setIsReadOnly(false)
+		getState(ctx).dispatcher.setIsReadOnly(true)
```

Comment:
So I see yield stuff below. Shouldn't you just mark read-only and not read-only strictly around validator function and query function invocation? (sorry if I am commenting on your changes too early)

Reply:
yes that's correct